### PR TITLE
Lessen dependencies on lodash. Shorten the code by 10%.

### PR DESCRIPTION
**Pull request by koresar**
Improve .gitignore. Add .editorconfig. Use latest lodash and moved out of dev. dependencies. Remove unused `dotty` dependency.

### CODE REVIEW

File: examples/compose.js [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
-import map from 'lodash/collection/map';
-import isUndefined from 'lodash/lang/isUndefined';
-import isFunction from 'lodash/lang/isFunction';
+import merge from 'lodash/merge';
+const isFunction = (obj) => typeof obj === 'function';
+const assign = Object.assign;
 
 const getDescriptorProps = (descriptorName, composables) => {
-  return map(composables, composable => {
+  return !composables ? undefined : composables.map(composable => {
```
Comment (unstoppablecarl) [inline examples/compose.js]: probably just my preference but this is a bit hard to read.

Comment (koresar) [inline examples/compose.js]: `null` is fine too. I just was trying to mimic the `lodash.map` behavior.
Change it to `null`?

Comment (troutowicz) [inline examples/compose.js]: Pretty subjective. :p

Comment (koresar) [inline examples/compose.js]: Agree to @troutowicz 

How about the PR in general, guys?

**Comment by unstoppablecarl** [comment]
Looks good. It may be nice to see more comments explaining things as it is an example.

**Comment by koresar** [comment]
good point. I'll create an issue @unstoppablecarl

**Comment by koresar** [comment]
The issue #65 got created. Will address the comments separately.

**Pull request merged**